src/core/thread/mle.cpp

```diff
@@ -1460,45 +1496,97 @@ void Mle::HandleAttachTimer(void)
             delay = kParentRequestRouterTimeout;
         }
 
+        delay += Random::GetUint32InRange(0, kParentRequestJitter);
+
         break;
 
     case kAttachStateParentRequestRouter:
-        mAttachState = kAttachStateParentRequestReed;
+        SetAttachState(kAttachStateParentRequestReed);
+        SendParentRequest(kParentRequestTypeRoutersAndReeds);
+        delay = kParentRequestReedTimeout + Random::GetUint32InRange(0, kParentRequestJitter);
+        break;
 
-        if (mParentCandidate.GetState() != Neighbor::kStateParentResponse)
+    case kAttachStateParentRequestReed:
+        shouldAnnounce = ShouldSendAnnounce();
+
+        if (shouldAnnounce)
         {
+            SetAttachState(kAttachStateAnnounce);
             SendParentRequest(kParentRequestTypeRoutersAndReeds);
-            delay = kParentRequestReedTimeout;
+            mAnnounceChannel = OT_RADIO_CHANNEL_MIN;
+            delay            = mAnnounceDelay;
             break;
         }
 
         // fall through
 
-    case kAttachStateParentRequestReed:
-        if (mParentCandidate.GetState() == Neighbor::kStateParentResponse &&
-            (mRole != OT_DEVICE_ROLE_CHILD || mReceivedResponseFromParent || mParentRequestMode == kAttachBetter) &&
-            SendChildIdRequest() == OT_ERROR_NONE)
+    case kAttachStateAnnounce:
+        if (shouldAnnounce)
         {
-            mAttachState = kAttachStateChildIdRequest;
-            delay        = kParentRequestReedTimeout;
+            bool didChannelWrap = SendOrphanAnnounce();
```

Comment:
At this level, all we care is whether or not we've sent an orphan on all the channels.  I think it would be cleaner if `SendOrphanAnnounce()` returned `otError` instead